### PR TITLE
Fix the bug jvm not exit when enableAuth

### LINKED ISSUE #2652: [BUG] If set spark.sql.auth.enable=true, the JVM does not exit when the sparksession is stoped
**Describe the bug**
image
As the title say

**What did you do**
image
It is beacuse the TiAuthorization use ScheduledExecutorService use default threadFactory not the Daemon Factory to build ScheduledThreadPool

### PR DESCRIPTION
**Pull request by wfxxh**
Set the Periodically update privileges from TiDB scheduler to Daemon thread

### What problem does this PR solve? 
close #2652 

### What is changed and how it works?
Set the thread to daemon

### CODE REVIEW

File: core/src/main/scala/com/pingcap/tispark/auth/TiAuthorization.scala [hunk 1]
```diff
@@ -41,7 +41,11 @@ case class TiAuthorization private (parameters: Map[String, String], tiConf: TiC
   private var jdbcClient: JDBCClient = _
 
   private val scheduler: ScheduledExecutorService =
-    Executors.newScheduledThreadPool(1)
+    Executors.newScheduledThreadPool(1, (r: Runnable) => {
```
Comment (shiyuhang0) [inline core/src/main/scala/com/pingcap/tispark/auth/TiAuthorization.scala:44]:
```suggestion
    Executors.newScheduledThreadPool(1, new ThreadFactoryBuilder().setDaemon(true).build());
```

Comment (wfxxh) [reply core/src/main/scala/com/pingcap/tispark/auth/TiAuthorization.scala]: shoud import com.google.common.util.concurrent.ThreadFactoryBuilder package

**Review by shiyuhang0 — APPROVED** [review]

**Review by xuanyu66 — APPROVED** [review]

**Comment by shiyuhang0** [comment]
merged. thank you for your contribute

**Comment by wfxxh** [comment]
It is my honour

**Pull request merged**